src/analytics/analytics-tracker-window.ts

```diff
@@ -1,9 +1,60 @@
+import { isFunction } from 'lodash';
+
 export interface AnalyticsTracker {
     track(step: string, data: any): void;
-    hit(hitName: string, data: any): void;
-    hasPayloadLimit(data: any): boolean;
 }
 
 export default interface AnalyticsTrackerWindow extends Window {
     analytics: AnalyticsTracker;
+    ga(command: string, eventName: string, payload: any): void;
 }
+
+export const sendGoogleAnalytics = (type: string, payload: any) => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    if (isFunction(_window?.ga)) {
+        _window.ga('send', type, {
+            ...payload,
+            nonInteraction: false,
+        });
+    }
+};
+
+export const isGoogleAnalyticsAvailable = () => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    return isFunction(_window?.ga);
+};
+
+/**
+ * Max size of the payload for the Google Analytics module
+ * if the limit will be succeeded, the GA throwing a silent error,
+ * and only in debug mode you can see it
+ */
+export const ANALYTICS_MAX_URI_LENGTH = 8096;
+
+export const isPayloadSizeLimitReached = (obj: any) => {
+    return serializeAnalyticsEventPayload(obj).length >= ANALYTICS_MAX_URI_LENGTH;
+};
+
+const serializeAnalyticsEventPayload = (obj: Record<string, any>): string => {
```

Comment:
Do you have to use any? Perhaps `unknown` or `string|number` is a better type.

Reply:
This function with recursive call and type always different.

Reply:
In that case, you can use `unknown`. You're already doing `typeof` checks so it should just work. The reason why I'm pushing people not to use `any` is to force engineers to do explicit type checkint whenever we're dealing with unknow types.

Reply:
done